matdeeplearn/trainers/property_trainer.py

```diff
@@ -37,22 +40,33 @@ def __init__(
             test_loader,
             loss,
             max_epochs,
+            max_checkpoint_epochs,
             identifier,
             verbosity,
+            save_dir,
+            checkpoint_dir,
         )
 
     def train(self):
+        epochs_to_run = (
+            self.max_checkpoint_epochs
+            if self.max_checkpoint_epochs > 0
+            else self.max_epochs
+        )
+
         if self.train_verbosity:
             logging.info("Starting regular training")
             logging.info(
-                f"running for  {self.max_epochs} epochs on {type(self.model).__name__} model"
+                f"running for {epochs_to_run} epochs on {type(self.model).__name__} model"
             )
 
         # Start training over epochs loop
         # Calculate start_epoch from step instead of loading the epoch number
         # to prevent inconsistencies due to different batch size in checkpoint.
+
         start_epoch = self.step // len(self.train_loader)
-        for epoch in range(start_epoch, self.max_epochs):
+
+        for epoch in range(start_epoch, epochs_to_run):
```

Comment:
So now it should be this, right? That way if you load from a checkpoint you're running `x` epochs more starting from the epoch you're currently at? Double check me on this, though
```suggestion
        for epoch in range(start_epoch, start_epoch + epochs_to_run):
```

Reply:
However, this would conflict with the idea of `max_epochs` if we decide to load from config but forget to specify the `max_checkpoint_epochs`. Instead, could we do the following to keep the loop unchanged?

```python
epochs_to_run = (
            self.max_checkpoint_epochs + start_epoch
            if self.max_checkpoint_epochs
            else self.max_epochs
        )
```

Reply:
Sure, I like that. How about you change the variable name though to `end_epoch`, because `epochs_to_run` implies that I want to run 100 epochs, but really we may want to be running epoch 200-299.